mava/systems/mat/anakin/mat.py

```diff
@@ -0,0 +1,647 @@
+        learner_state = LearnerState(params, opt_state, key, env_state, last_timestep)
+
+        metric = traj_batch.info
+
+        return learner_state, (metric, loss_info)
+
+    def learner_fn(learner_state: LearnerState) -> ExperimentOutput[LearnerState]:
+        """Learner function.
+
+        This function represents the learner, it updates the network parameters
+        by iteratively applying the `_update_step` function for a fixed number of
+        updates. The `_update_step` function is vectorized over a batch of inputs.
+
+        Args:
+            learner_state (NamedTuple):
+                - params (Params): The initial model parameters.
+                - opt_state (OptStates): The initial optimiser state.
+                - key (chex.PRNGKey): The random number generator state.
+                - env_state (LogEnvState): The environment state.
+                - timesteps (TimeStep): The initial timestep in the initial trajectory.
+        """
+
+        batched_update_step = jax.vmap(_update_step, in_axes=(0, None), axis_name="batch")
+
+        learner_state, (episode_info, loss_info) = jax.lax.scan(
+            batched_update_step, learner_state, None, config.system.num_updates_per_eval
+        )
+        return ExperimentOutput(
+            learner_state=learner_state,
+            episode_metrics=episode_info,
+            train_metrics=loss_info,
+        )
+
+    return learner_fn
+
+
+def learner_setup(
+    env: Environment, keys: chex.Array, config: DictConfig
+) -> Tuple[LearnerFn[LearnerState], Any, LearnerState]:
+    """Initialise learner_fn, network, optimiser, environment and states."""
+    # Get available TPU cores.
+    n_devices = len(jax.devices())
+
+    # Get number of agents.
+    config.system.num_agents = env.num_agents
+
+    # PRNG keys.
+    key, actor_net_key, _ = keys
+
+    # Initialise observation: Obs for all agents.
+    init_x = env.observation_spec().generate_value()
+    init_x = tree.map(lambda x: x[None, ...], init_x)
+
+    _, action_space_type = get_action_head(env)
+
+    if action_space_type == "discrete":
+        init_action = jnp.zeros((1, config.system.num_agents), dtype=jnp.int32)
+    elif action_space_type == "continuous":
+        init_action = jnp.zeros((1, config.system.num_agents, env.action_dim), dtype=jnp.float32)
+    else:
+        raise ValueError("Invalid action space type")
+
+    # Define network and optimiser.
+    actor_network = MultiAgentTransformer(
+        obs_dim=init_x.agents_view.shape[-1],
+        action_dim=env.action_dim,
+        n_block=config.network.actor_network.n_block,
+        n_head=config.network.actor_network.n_head,
+        n_embd=config.network.actor_network.n_embd,
+        n_agent=config.system.num_agents,
+        use_rmsnorm=config.network.use_rmsnorm,
+        use_swiglu=config.network.use_swiglu,
+        action_space_type=action_space_type,
+    )
```

Comment:
Definitely think initializing with a config would be nice here

Reply:
We would need to agree on what this config looks like, because Sable doesn't use a network config like you'd expect either.

Reply:
Please see the net_config update :)